File: Numeros Pseudo-Aleatorios/IBM RANDU/randu.dart
/*
   PRNG .......: RANDU
   Ano ........: Decada de 1960
   Autor ......: IBM
*/

import 'dart:io';

class PRNG {
  static int RANDU(int seed) {
    return (65539 * seed) % 2147483648;
  }
}

void main() {

  int Xn1 = 1234;
  int Xn2 = DateTime.now().millisecondsSinceEpoch ~/ 1000;

  print("GERADOR DE NUMEROS PSEUDO ALEATORIOS (GNA)");
  print("PSEUDORANDOM NUMBER GENERATOR (PRNG)");
  print("PADRAO: RANDU");
  print("==========================================");
  print("");
  print("    Semente: 1234        Semente: TIME");
  print("    -------------        -------------");

  for (int i = 1; i <= 10; i++) {
    Xn1 = PRNG.RANDU(Xn1);
    Xn2 = PRNG.RANDU(Xn2);
    print("${Xn1.toString().padLeft(17)}        ${Xn2.toString().padLeft(13)}");
  }

}

File: Numeros Pseudo-Aleatorios/IBM RANDU/randu.c
/* 
   PRNG .......: RANDU
   Ano ........: Decada de 1960
   Autor ......: IBM
*/

#include <stdio.h>
#include <time.h>

long long int RANDU(long long int *seed) {
    *seed = (65539 * (*seed)) % 2147483648;
    return *seed;
}

int main(void) {

    long long int Xn1, Xn2;
    int i;
    
    Xn1 = 1234;
    Xn2 = (long long int)time(NULL);
    
    printf("GERADOR DE NUMEROS PSEUDO ALEATORIOS (GNA)\n");
    printf("PSEUDORANDOM NUMBER GENERATOR (PRNG)\n");
    printf("PADRAO: RANDU\n");
    printf("==========================================\n\n");
    printf("    Semente: 1234        Semente: TIME\n");
    printf("    -------------        -------------\n");
    
    for (i = 1; i <= 10; i++) {
        printf("%17lld        %13lld\n", RANDU(&Xn1), RANDU(&Xn2));
    }
    
    return 0;
}
